authentication/ldap_authentication_provider.py

```diff
@@ -12,35 +12,156 @@
 #  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  * See the License for the specific language governing permissions and
 #  * limitations under the License.
-
+import logging
 
 import ldap
 from flask_securest.authentication_providers.abstract_authentication_provider \
     import AbstractAuthenticationProvider
 from flask_securest import utils
+from flask_securest import constants
+from flask_securest.exceptions import AuthenticationException
+
+DN_ATTRIBUTE_NAME = 'DN'
+DEFAULT_AUTH_ATTR_NAME = 'uid'
+
+REQ_SEARCH_FIELDS = ('base_dn', 'admin_user_id', 'admin_password')
 
 
 class LDAPAuthenticationProvider(AbstractAuthenticationProvider):
-    def __init__(self, directory_url):
-        self.directory_url = directory_url
 
+    _logger = logging.getLogger(constants.FLASK_SECUREST_LOGGER_NAME)
+
+    def __init__(self, ldap_url, search_properties=None):
+        self.ldap_url = ldap_url
+        self.search_props = search_properties
+        self.validate_search_props()
+
+    def validate_search_props(self):
+        if self.search_props:
+            valid = all(test in self.search_props for test in
```

Comment:
1. `test` should be renamed to `value` or `element`
2. the condition is wrong. an empty value for a field in the search properties will cause `test in self.search_props` to return `True`, when in fact it is not valid. It should be:
valid = all(search_props.get(value, None) for value in REQ_SEARCH_FIELDS)
3. the valid variable is redundant. you can just put the condition into line 43 like this:
if not all(search_props.get(value, None) for value in REQ_SEARCH_FIELDS):